update to 0.16.0

boldorider4 opened the pull request:
* [x] checked license
* [x] updated dependencies with [upstream](https://github.com/arviz-devs/arviz/blob/main/requirements.txt)
* [x] updated about section

This version of `arviz` makes use of one of two possible packages that are similar implementations of the same api:
* `h5netcdf`
* `netcdf4`

while in the past `netcdf4` has been preferred, in this iteration `h5netcdf` is set as the default backing implementation.
This results in dropping `netcdf4` as one of the dependencies (which is in fact also not referenced by the [upstream requirements](https://github.com/arviz-devs/arviz/blob/main/requirements.txt)). While supporting both may expand the number of supported platforms, `h5netcdf` allows to build all that `netcdf4` does and then some. Hence `netcdf4` has been dropped.

boldorider4 committed:
update recipe to 0.15.1

boldorider4 committed:
update hash

boldorider4 committed:
align dependencies with upstream requirements

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_09wvd4drvq/clone/recipe/meta.yaml:21: missing_wheel: For pypi packages, wheel should be present in the host section
WARNING: /tmp/abs_09wvd4drvq/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_09wvd4drvq/clone/recipe/meta.yaml:36: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_d03ii0cuqs/clone/recipe/meta.yaml:21: missing_wheel: For pypi packages, wheel should be present in the host section
WARNING: /tmp/abs_d03ii0cuqs/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_d03ii0cuqs/clone/recipe/meta.yaml:36: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

boldorider4 committed:
address dependency issues

boldorider4 committed:
complete tests with pip check

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_7ac4w_430n/clone/recipe/meta.yaml:22: missing_wheel: For pypi packages, wheel should be present in the host section
ERROR: /tmp/abs_7ac4w_430n/clone/recipe/meta.yaml:22: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_7ac4w_430n/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_7ac4w_430n/clone/recipe/meta.yaml:32: missing_python: For pypi packages, python should be present in the host and run sections. Missing in run
Errors were found

</details>

boldorider4 committed:
skip python 3.8

boldorider4 committed:
fix typo

boldorider4 committed:
fix pinning of numpy as compatible pin does not work

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_3bk31cdn1m/clone/recipe/meta.yaml:32: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_3bk31cdn1m/clone/recipe/meta.yaml:34: missing_python: For pypi packages, python should be present in the host and run sections. Missing in run
Errors were found

</details>

boldorider4 committed:
support both netcdf4 and h5netcdf implementations

tests seem to require both

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_87lh4gduys/clone/recipe/meta.yaml:33: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_87lh4gduys/clone/recipe/meta.yaml:35: missing_python: For pypi packages, python should be present in the host and run sections. Missing in run
Errors were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_f6tmtwcpvl/clone/recipe/meta.yaml:33: python_build_tool_in_run: The python build tool packaging is in run depends
ERROR: /tmp/abs_f6tmtwcpvl/clone/recipe/meta.yaml:35: missing_python: For pypi packages, python should be present in the host and run sections. Missing in run
Errors were found

</details>

boldorider4 committed:
get rid of dependency on netcdf4

h5netcdf is already providing the necessary backend and is preferred
to netcdf4.

boldorider4 committed:
address linter problems

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_fc9xkzkz5z/clone/recipe/meta.yaml:33: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

boldorider4 committed:
build for every platform individually

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_0cnbsjbhx5/clone/recipe/meta.yaml:32: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

ryanskeith commented on recipe/meta.yaml:
```diff
@@ -1,35 +1,45 @@
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  # skipping 3.8 because of missing xarray-einstats for 3.8
+  skip: True  # [py==3.8]
```
```suggestion
```

ryanskeith commented on recipe/meta.yaml:
```diff
@@ -1,35 +1,45 @@
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  # skipping 3.8 because of missing xarray-einstats for 3.8
+  skip: True  # [py==3.8]
+  skip: True  # [py<=2.7]
```
```suggestion
  skip: True  # [py<39]
```

ryanskeith replied:
Just a simplification.

boldorider4 replied:
I thought these kinds of restrictions that are coming from the upstream project had to be retained even though we don't support `py<39`

ryanskeith commented on recipe/meta.yaml:
```diff
@@ -1,35 +1,45 @@
-    - python >=3
+    - python
     - pip
+    - setuptools
```
```suggestion
    - setuptools >=60.0.0
```
According to this https://github.com/arviz-devs/arviz/blob/f80265030f9662d1e31a680ad05cc58e6f3bf9eb/requirements.txt#L1

boldorider4 replied:
In the host section we are required to use exact pinning (no ranges).
Furthermore, we want python build tools to lose the pinning altogether, Linter would also complain.

M-Waszkiewicz-Anaconda replied:
I think we should follow upstream the same way as we follow with supported python version, we can discuss it in the meeting

ryanskeith reviewed (commented):
I would definitely update the setuptools requirement.  Also, making sure the tests pass is important.

boldorider4 committed:
simplify python version skipping

Co-authored-by: Ryan <rkeith@anaconda.com>

boldorider4 committed:
simplify python version skipping

Co-authored-by: Ryan <rkeith@anaconda.com>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_fdid7fkiot/clone/recipe/meta.yaml:31: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

boldorider4 committed:
fix duplicated switch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_99ubv4owsx/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

lorepirri reviewed (approved):
LGTM

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -1,35 +1,43 @@
 {% set name = "arviz" %}
-{% set version = "0.11.2" %}
+{% set version = "0.15.1" %}
```
v0.16.0 was released 3 days ago. It dropped `py<39` support and changed `numpy` pinning in runtime. Could we build the latest version?

M-Waszkiewicz-Anaconda committed:
bumpup version to 0.16.0

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_ab6gk6nts_/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

M-Waszkiewicz-Anaconda reviewed (approved):
LGTM

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_6anlaqv3vn/clone/recipe/meta.yaml:30: python_build_tool_in_run: The python build tool packaging is in run depends
Warnings were found

</details>

lorepirri commented:
I've deleted the branch and updated the settings to do so automatically after the merge
